webapp/channels/src/components/setting_item.tsx

```diff
@@ -41,37 +41,43 @@ type Props = {
     collapsedEditButtonWhenDisabled?: ReactNode;
 }
 
-export default class SettingItem extends React.PureComponent<Props> {
-    minRef: RefObject<SettingItemMinComponent>;
+const SettingItem = ({
+    active,
+    areAllSectionsInactive,
+    max,
+    title,
+    updateSection,
+    describe,
+    section,
+    isDisabled,
+    collapsedEditButtonWhenDisabled,
+}: Props) => {
+    const minRef = useRef<SettingItemMinComponent>(null);
 
-    constructor(props: Props) {
-        super(props);
+    useEffect(() => {
```

Comment:
In the original logic, we only do this on update. We should already have a `useDidUpdate` hook.